### PR TITLE
.Net Processes - Use JSON Schema Generation from Type

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Fixes: https://github.com/microsoft/semantic-kernel/issues/9298

### Description

Following up on @esttenorio's  suggestion to use schema generation.  

- Adding new package dependency: `JsonSchema.Net.Generation`
- We already reference `JsonSchema.Net`, but it was a stale version.  Bumped to current
- Updated sample

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by stephentoub** [comment]
Can we use the same JSON schema generation support that's used by the core of SK and that doesn't depend on JsonSchema.Net?

**Comment by crickman** [comment]
> Can we use the same JSON schema generation support that's used by the core of SK and that doesn't depend on JsonSchema.Net?

@stephentoub - I could've sworn that `Functions.OpenAPI` was actively using the `JsonSchema.Net` dependency.  In anycase, I must've missed the SK generator support in my survey.  I'll look again.  Any pointers appreciated.

**Comment by stephentoub** [comment]
https://github.com/microsoft/semantic-kernel/blob/b55cb8a0e10de5f33c50be8016d28be802b465f1/dotnet/src/InternalUtilities/src/Schema/KernelJsonSchemaBuilder.cs#L10

File: dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs [hunk 1]
```diff
@@ -3,6 +3,8 @@
 using System.ComponentModel.DataAnnotations;
```
Comment (esttenorio) [inline dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs:3]: do all the bot warnings get fix with removing this import? `System.ComponentModel.DataAnnotations;`

**Comment by crickman** [comment]
> https://github.com/microsoft/semantic-kernel/blob/b55cb8a0e10de5f33c50be8016d28be802b465f1/dotnet/src/InternalUtilities/src/Schema/KernelJsonSchemaBuilder.cs#L10

Thanks

File: dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs (hunk 1 above)
Comment (crickman) [reply dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs:3]: Yes, I believe so.  I thought I had already removed it, but must have not commited.  Been distracted on the investigating the alternate approach.  Updating namespace now to stabilize this version.

**Review by esttenorio — APPROVED** [review]

**Comment by crickman** [comment]
> https://github.com/microsoft/semantic-kernel/blob/b55cb8a0e10de5f33c50be8016d28be802b465f1/dotnet/src/InternalUtilities/src/Schema/KernelJsonSchemaBuilder.cs#L10

Ok, I got this working using the utility we copied with a tiny bit of hackery to include the `additionalProperties` property.  Once we shift to `System.Text.Json` v9.0.0, the `additionalProperties` is handled by the generator; although, the rest of the code stays about the same (i.e. the huge block of copied code for the `TransformSchemaNode` option - https://devblogs.microsoft.com/dotnet/system-text-json-in-dotnet-9).

**Review by alliscode — APPROVED** [review]

**Pull request merged**